common/lib/xmodule/xmodule/graders.py

```diff
@@ -81,25 +82,58 @@ def float_sum(iterable):
 
 def aggregate_scores(scores):
     """
-    scores: A list of ScoreBase objects
+    scores: A list of ProblemScore objects
     returns: A tuple (all_total, graded_total).
-        all_total: A ScoreBase representing the total score summed over all input scores
-        graded_total: A ScoreBase representing the score summed over all graded input scores
+        all_total: An AggregatedScore representing the total score summed over all input scores
+        graded_total: An AggregatedScore representing the score summed over all graded input scores
     """
-    total_correct_graded = float_sum(score.earned for score in scores if score.graded)
-    total_possible_graded = float_sum(score.possible for score in scores if score.graded)
-    any_attempted_graded = any(score.attempted for score in scores if score.graded)
+    def _iter_graded(scores):
```

Comment:
For my own edification, when *do* we want to create internal functions? I can't think of a case where I'd rather that than a small method outside the "main" method - and very small cases are already better handled by creating a `lambda`, yes?